internal_packages/keybase/lib/private-key-popover.cjsx

```diff
@@ -0,0 +1,138 @@
+{React, Actions, AccountStore} = require 'nylas-exports'
+{remote} = require 'electron'
+Identity = require './identity'
+PGPKeyStore = require './pgp-key-store'
+PassphrasePopover = require './passphrase-popover'
+_ = require 'underscore'
+fs = require 'fs'
+pgp = require 'kbpgp'
+
+module.exports =
+class PrivateKeyPopover extends React.Component
+  constructor: ->
+    @state = {
+      selectedAddress: "0"
+      keyBody: ""
+      paste: false
+      import: false
+      validKeyBody: false
+    }
+
+  @propTypes:
+    addresses: React.PropTypes.array
+
+  render: =>
+    errorBar = <div className="invalid-key-body">Invalid key body.</div>
+    keyArea = <textarea value={@state.keyBody || ""} onChange={@_onKeyChange} placeholder="Paste in your PGP key here!"/>
+
+    saveBtnClass = if !(@state.validKeyBody) then "btn modal-done-button btn-disabled" else "btn modal-done-button"
+    saveButton = <button className={saveBtnClass} disabled={!(@state.validKeyBody)} onClick={@_onDone}>Save</button>
+
+    <div className="private-key-popover" tabIndex=0>
+      <span key="title" className="picker-title"><b>No PGP private key found.<br/>Add a key for {@_renderAddresses()}</b></span>
+      <div className="key-add-buttons">
+        <button className="btn btn-toolbar paste-btn" onClick={@_onClickPaste}>Paste in a Key</button>
+        <button className="btn btn-toolbar import-btn" onClick={@_onClickImport}>Import from File</button>
+      </div>
+      {if (@state.import or @state.paste) and !@state.validKeyBody and @state.keyBody != "" then errorBar}
+      {if @state.import or @state.paste then keyArea}
+      <div className="picker-controls">
+        <div style={{width: 80}}><button className="btn modal-cancel-button" onClick={() => Actions.closePopover()}>Cancel</button></div>
+        <button className="btn modal-prefs-button" onClick={@_onClickAdvanced}>Advanced</button>
+        <div style={{width: 80}}>{saveButton}</div>
+      </div>
+    </div>
+
+  _renderAddresses: =>
+    signedIn = _.pluck(AccountStore.accounts(), "emailAddress")
+    suggestions = _.intersection(signedIn, @props.addresses)
+
+    if suggestions.length == 1
+      addresses = <span>{suggestions[0]}.</span>
+    else if suggestions.length > 1
+      options = suggestions.map((address) => <option value={suggestions.indexOf(address)} key={suggestions.indexOf(address)}>{address}</option>)
+      addresses =
```

Comment:
Hmm? It gets used up in _render when _renderAddresses() is called. That dropdown definitely shows up...